### DISCUSSION TITLE
Exception for good old `Provider` from the "generated providers should only depend on other generated providers" rule

### DISCUSSION THREAD

**Discussion by GokulDas027** [rrousselGit/riverpod#3166] [General]
I had been using the basic `Provider` for dependency injection, which is simple, straight forward and generators would be an overkill.

![image](https://github.com/rrousselGit/riverpod/assets/22936485/33afc9c4-211c-434a-a24f-1028eaf8a58d)

Now, on calling the provider from a generated provider, riverpod_lint is saying
> Generated providers should only depend on other generated providers. Failing to do so may break rules such as "provider_dependencies".

How will the future be for the manually declared `provider` in v3 or forward, considering it is the simplest and basic provider, could be kept as it is and excepted from the generated provider only rule.

I've read [here](https://github.com/rrousselGit/riverpod/discussions/1725#discussioncomment-3797504) from @rrousselGit (about StateProvider), the rule is mainly intented as riverpod is heading the metaprogramming direction further, or is there any runtime issues on for generated code depending on manually declared providers.

Request to keep Provider as such and exception from the rule.

**Comment by rrousselGit** [comment]
That's unlikely to be realistic. There are technical reasons why it works only with the code-generator.

And the non-codegenerator syntax is bound to disappear in a distant future
